frontend/simple/views/CreateGroup.vue

```diff
@@ -223,18 +221,18 @@ export default {
           founderIdentityContractId: this.$store.state.loggedIn.identityContractId
         })
         const hash = entry.toHash()
-        // TODO: convert this to SBL
-        Vue.events.$once(hash, (contractId, entry) => {
-          this.$store.commit('setCurrentGroupId', hash)
-          // Take them to the invite group members page.
-          this.$router.push({path: '/invite'})
-        })
-        // TODO: convert this to SBL
-        await backend.publishLogEntry(hash, entry)
-        // add to vuex and monitor this contract for updates
+
+        await sbp('transactions/v1/run', 'Create a Group', false, [
+          { execute: 'setInScope', args: { contractId: hash, entry } },
+          { execute: 'backend/v1/publishLogEntry', description: 'Publish Group Contract', args: { contractId: 'contractId', entry: 'entry' } }
+        ])
```

Comment:
OK, so after thinking/working on various issues, especially on how to design an API that [meets these requirements](https://github.com/okTurtles/group-income-simple/pull/320/files#r137940375) here's what I came up with:

```js
const hash = entry.toHash()
await sbp('transactions/v1/enqueue', 'Create a Group', [
  {
    skipCondition: ['db/v1/hasHash', hash],
    beforeTransaction: ['util/v1/batch',
      // to disable some UI elements
      [
        'vuex/v1/store/commit',
        'setAppState', 
        ['publishingLogEntry']
      ],
      // listen for server to echo back event and only
      // then do we set the currentGroupId & move to invite
      [
        'vuex/v1/events/once', hash, [
          'util/v1/batch', [
            ['vuex/v1/store/commit', 'setCurrentGroupId', hash],
            ['router/v1/push', {path: '/invite'}]
          ]
        ]
      ]
    ],
    transaction: [
      'contracts/v1/publishLogEntry',
      {
        type: 'GroupContract',
        data: entry.data
      }
    ],
    onSuccess: ['util/v1/batch',
      [
        'vuex/v1/store/dispatch',
        'syncContractWithServer',
        hash
      ],
      [
        'vuex/v1/store/commit',
        'unsetAppState',
        ['publishingLogEntry']
      ]
    ],
    onError: 'notifications/v1/error'
  }
])
```

Note it also addresses the next comment/issue below:

Reply:
Note that this example uses a modified version of `'vuex/v1/events/once'` that also supports strings as selectors handlers (in addition to normal functions)

Reply:
To summarize today's call:

- We've decided that perhaps it's overkill to wait for a transaction to be processed back through `handleEvent` to consider it safe to remove from the queue, and maybe be sufficient to just consider a `200 OK` message from the server to pop a transaction from the queue
- This means, at least for a first-draft, it might not be necessary to implement `skipCondition` or `beforeTransaction`... We'll see how it goes and re-add them later if necessary